ush/ocnpost.py

```diff
@@ -0,0 +1,646 @@
+            exit()
+
+    output_masks=False
+    methods=['bilinear','conserve']
+    dsttype=['rect.']
+    regridtype='manual' #manual or auto
+
+    #variables to be regridded with the native tripole stagger location
+    #and dimensionality 
+    #first BM contained only field "mld", which was actually ePBL
+    #the remaining BMs contain ePBL, MLD_003 and MLD_0125 
+    #the following NCO command will be issued at the end 
+    #to rename the variable mld to ePBL if the variable mld is found
+
+    ncocmd=['ncrename -O -v MLD_003,mld']
+
+    #If a parm file is given
+    if len(sys.argv) == 6:
+        print('Processing variables from parm')
+        with open(parmfile) as f3:
+            parmvars = f3.read().splitlines()
+
+        nvarsall=len(parmvars)
+        nvars=0
+        nvpairs=0
+        vartype=np.zeros(shape=(nvarsall),dtype=list)
+        for nva in range(0,nvarsall):
+            if ',' in parmvars[nva]:
+                vartype[nva]='vector'
+                nvpairs=nvpairs+1
+            else:
+                vartype[nva]='scalar'
+                nvars=nvars+1
+
+        parmvars_s=np.array(parmvars)[np.argwhere(vartype=='scalar')]
+        parmvars_v=np.array(parmvars)[np.argwhere(vartype=='vector')]
+
+        varlist=np.zeros(shape=(nvars,4),dtype=list)
+        for nv in range(0,nvars):
+            nvarparm1=np.char.split(parmvars_s[nv],':')
+            varlist[nv,:]=nvarparm1[0]           
+
+        veclist=np.zeros(shape=(nvpairs,4,2),dtype=list)
+        for nv in range(0,nvpairs):
+            nvectparm1=np.char.split(parmvars_v[nv],':')
+            nvectparm2=nvectparm1[0]      
+            for nvp in range(0,4):
+                veclist[nv,nvp,:]=nvectparm2[nvp].split(',')
+
+    #----------------------------------------------------------------------
+    # make a list of the directories and files from the run 
+    #----------------------------------------------------------------------
+
+    #specify input file name and input directory
+    print('Extracting Input Data')
+    ocnf=nc.Dataset(infiles1[0])  #Read using nc
+    
+
+    #Get a list of available variables from netcdf file
+    ncvarlist=ocnf.variables.keys()  #Get list of variables using nc
+
+    #Check which model is to be post-processed 
+    #by searching for a key variable specific to that model
+    if 'SST' in ncvarlist:
+        model='MOM'
+    elif 'Tsfc_d' in ncvarlist or 'Tsfc_h' in ncvarlist:
+        model='CICE'
+        if 'Tsfc_d' in ncvarlist:
+            samplevar='Tsfc_d'
+        if 'Tsfc_h' in ncvarlist:
+            samplevar='Tsfc_h' 
+    else:
+        print('Product not supported. Exiting program...')
+        exit()
+    print('Post-processing '+str(model))
+
+    #Extract Source Grid coordinates
+    if model == 'MOM':
+        dimxh=ocnf.dimensions['xh'].size
+        dimyh=ocnf.dimensions['yh'].size
+    if model == 'CICE':
+        dimxh=ocnf.dimensions['ni'].size
+        dimyh=ocnf.dimensions['nj'].size
```

Comment:
Can `model` be both `MOM` and `CICE` at the same time?

Reply:
This program can only process MOM and CICE one at a time, therefore `model` can only be either `MOM` or `CICE`.

Reply:
In that case, this program needs to be written in a way that leverages OOP.

Reply:
The program should be agnostic to the model. All it requires if the ESMF coefficients and a FORTRAN bit to remap the variables. ESMF is also agnostic to the model. All it cares about is mapping points from a source grid to a destination grid. The only scenario that I can think of in which you would need to specify the model is if you are generating a MOM6 or CICE tri-polar projection from scratch.